src/fomo.jl

```diff
@@ -0,0 +1,399 @@
+export fomo, FomoSolver, FoSolver, R2, tr_step, r2_step
+
+abstract type AbstractFirstOrderSolver <: AbstractOptimizationSolver end
+
+abstract type AbstractFomoMethod end
+struct tr_step   <: AbstractFomoMethod end
+struct r2_step   <: AbstractFomoMethod end
+
+"""
+    fomo(nlp; kwargs...)
+    R2(nlp; kwargs...)
+
+A First-Order with MOmentum (FOMO) model-based method for unconstrained optimization. Supports quadratic regularization and trust region methods.
+
+For advanced usage, first define a `FomoSolver` to preallocate the memory used in the algorithm, and then call `solve!`:
+
+    solver = FomoSolver(nlp)
+    solve!(solver, nlp; kwargs...)
+
+**Quadratic Regularization (R2)**: if the user do not want to use momentum (`β` = 0), it is recommended to use the memory-optimized `R2` method.
+For advanced usage:
+
+    solver = FoSolver(nlp)
+    solve!(solver, nlp; kwargs...)
+Extra keyword arguments `σmin` is accepted (`αmax` will be set to `1/σmin`).
+
```

Comment:
Explain here how to momentum is defined and updated, to avoid having to exlpain in the kwargs section below. Number equations so you can refer to them.